plugins/kubernetes/app/models/watchers/base_cluster_watcher.rb

```diff
@@ -8,8 +8,9 @@ class BaseClusterWatcher
 
     finalizer :stop_watching
 
-    def initialize(watch_stream)
-      @watch_stream = watch_stream
+    def initialize(args)
+      @cluster = args[:cluster]
+      @watch_stream = args[:watch_stream]
```

Comment:
Ok, I see now. Derived `initialize` turns the cluster into a hash with the cluster & stream and passes it to superclass. Maybe composition in this case would be better than inheritance, we seem to be stretching thing a bit.